### PR TITLE
Fix connection timeout

### LINKED ISSUE #414: No response from server
These days, horta very often generates messages like this:

```
ERROR r.o.c.h.p.jabber.JabberProtocol - Cannot join room codingteam@conference.jabber.ru, retrying in 5 seconds
org.jivesoftware.smack.XMPPException: No response from server.
        at org.jivesoftware.smackx.muc.MultiUserChat.join(MultiUserChat.java:498) ~[horta-hell.jar:0.17.1]
        at org.jivesoftware.smackx.muc.MultiUserChat.join(MultiUserChat.java:392) ~[horta-hell.jar:0.17.1]
        at ru.org.codingteam.horta.protocol.jabber.JabberProtocol$$anonfun$receive$1$$anonfun$2.apply(JabberProtocol.scala:69) ~[horta-hell.jar:0.17.1]
        at ru.org.codingteam.horta.protocol.jabber.JabberProtocol$$anonfun$receive$1$$anonfun$2.apply(JabberProtocol.scala:63) ~[horta-hell.jar:0.17.1]
        at scala.util.Try$.apply(Try.scala:192) ~[horta-hell.jar:0.17.1]
        at ru.org.codingteam.horta.protocol.jabber.JabberProtocol$$anonfun$receive$1.applyOrElse(JabberProtocol.scala:63) ~[horta-hell.jar:0.17.1]
        at akka.actor.Actor$class.aroundReceive(Actor.scala:465) ~[horta-hell.jar:0.17.1]
        at ru.org.codingteam.horta.protocol.jabber.JabberProtocol.aroundReceive(JabberProtocol.scala:19) ~[horta-hell.jar:0.17.1]
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516) [horta-hell.jar:0.17.1]
        at akka.actor.ActorCell.invoke(ActorCell.scala:487) [horta-hell.jar:0.17.1]
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238) [horta-hell.jar:0.17.1]
        at akka.dispatch.Mailbox.run(Mailbox.scala:220) [horta-hell.jar:0.17.1]
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393) [horta-hell.jar:0.17.1]
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260) [horta-hell.jar:0.17.1]
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339) [horta-hell.jar:0.17.1]
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979) [horta-hell.jar:0.17.1]
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107) [horta-hell.jar:0.17.1]
12:16:58.216 [CodingteamSystem-akka.actor.default-dispatcher-4] INFO  r.o.c.h.p.jabber.MucMessageHandler - Room codingteam@conference.jabber.ru stopped
```

After that it could drop the room and tries to reconnect without success. I suspect there's something on the server side: http://stackoverflow.com/questions/13290499/could-not-connect-to-the-xmpp-server-via-smack-no-response-from-server

Need to investigate.

### PR DESCRIPTION
**Pull request by ForNeVeR**
This PR adds a configurable timeout that will be used for both XMPP stuff and socket timeouts. This fixes the bug in my environment if I use 30 sec timeout.

Closes #414.

### CODE REVIEW

**Comment by ForNeVeR** [comment]
Currently fixing the tests.

**Comment by ForNeVeR** [comment]
@rexim please review!

File: src/main/scala/ru/org/codingteam/horta/configuration/Configuration.scala [hunk 1]
```diff
@@ -41,6 +44,10 @@ object Configuration {
   lazy val login = properties.getProperty("login")
   lazy val password = properties.getProperty("password")
   lazy val server = properties.getProperty("server")
+  lazy val xmppTimeout = {
+    val ms = Option(properties.getProperty("xmpp.timeout_ms")).getOrElse("5000")
+    Integer.parseInt(ms).milliseconds
```
Comment (rexim) [inline src/main/scala/ru/org/codingteam/horta/configuration/Configuration.scala]: Why not something like?

```scala
Option(properties.getProperty("xmpp.timeout_ms"))
  .map(Integer.parseInt(_))
  .getOrElse(5000)
  .milliseconds
```

Doesn't invoke `Integer.parseInt()` if there is no `xmpp.timeout_ms`

Furthermore `Integer.parseInt` can fail with `NumberFormatException`. Are we gonna ignore that fact for now?

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/JabberProtocol.scala [hunk 2]
```diff
@@ -151,6 +154,22 @@ class JabberProtocol() extends Actor with ActorLogging {
...
     val configuration = new ConnectionConfiguration(server)
     configuration.setReconnectionAllowed(false)
+    configuration.setSocketFactory(new SocketFactory {
+      override def createSocket(host: String, port: Int): Socket = {
+        val socket = new Socket(host, port)
+        socket.setSoTimeout(Configuration.xmppTimeout.toMillis.toInt)
+        socket
+      }
+
+      // These aren't used by Smack, no point to implement them
```
Comment (rexim) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/JabberProtocol.scala]: Is this officially documented somewhere?

File: src/main/scala/ru/org/codingteam/horta/configuration/Configuration.scala (hunk 1 above)
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/configuration/Configuration.scala]: Nice, will change to your proposal.

> `Integer.parseInt` can fail with `NumberFormatException`. Are we gonna ignore that fact for now?

Yes.

File: src/main/scala/ru/org/codingteam/horta/protocol/jabber/JabberProtocol.scala (hunk 2 above)
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/protocol/jabber/JabberProtocol.scala]: Nope, I've just grokked through their source.

**Comment by ForNeVeR** [comment]
Thanks for the review.

**Pull request merged**